MoreLinq.Test/ScanByTest.cs

```diff
@@ -48,10 +50,10 @@ public void ScanBy()
             };
 
             var result =
-                    source.ScanBy(
-                        item => item.First(),
-                        key => (Element: default(string), Key: key, State: key - 1),
-                        (state, key, item) => (item, char.ToUpperInvariant(key), state.State + 1));
+                source.ScanBy(
+                    item => item.First(),
+                    key => (Element: string.Empty, Key: key, State: key - 1),
```

Comment:
Using `default(string?)` (rightfully) changes the element type to `string?`, which would then have to be handled below. `string.Empty` keeps type as `string` instead of `string?`.